src/Google.Cloud.Functions.Framework/FunctionException.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright 2020, Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Net;
+
+namespace Google.Cloud.Functions.Framework
+{
+    // More decisions and considerations:
+    // - Another message to be set on the response? (As content? Something else?)
```

Comment:
I don't like this. It might be used to write random content, specially if success status codes are allowed.

Reply:
Well users can already write whatever content they want... it just gives an easy way to give callers a hint as to what's wrong (e.g. which required parameter is missing)

Reply:
Yes, that's true, I guess what I really don't like is the combination of this and being able to set success status code from the exception. I can imagine it being used to control flow and that makes me icky.
I like a lot what MVC has with the IActionResult and all sorts of derived types, even types for bad status code like UnauthorizedResult, NotFoundResult, etc. It gave the opportunity to cleanly control for those cases I think.